app/components/articleSearch/components/noResult/noResultContent.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from "react";
+import { withStyles } from "../../../../helpers/withStylesHelper";
+import FilterResetButton from "../../../filterContainer/filterResetButton";
+const styles = require("./noResult.scss");
+
+interface NoResultContentProps {
+  hasEmptyFilter: boolean;
+  doiPatternMatched: boolean;
+  doi: string | null;
+  searchText: string | null;
+  handleSetIsOpen: () => void;
+}
+
+function disabledFilterMessage(hasEmptyFilter: boolean) {
```

Comment:
in this case, you can use `const DisabledFilterMessage: FunctionComponent<{hasEmptyFilter: boolean}> = () => React.memo{}` something like `pureComponent`.